micrometer-commons/src/main/java/io/micrometer/common/annotation/AnnotationHandler.java

```diff
@@ -59,6 +57,8 @@ public class AnnotationHandler<T> {
 
     private final BiFunction<Annotation, Object, KeyValue> toKeyValue;
 
+    private final BiPredicate<Annotation, Object> validToAdd;
```

Comment:
I think we should not do this, see my comment on `ObservedKeyValueAnnotationHandler`.